core/embed/sys/power_manager/stm32u5/power_manager_internal.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <sys/power_manager.h>
+#include <sys/systimer.h>
+#include <trezor_types.h>
+
+#include "../fuel_gauge/fuel_gauge.h"
+#include "../npm1300/npm1300.h"
+#include "../stwlc38/stwlc38.h"
+
+// Power manager thresholds & timings
+#define PM_TIMER_PERIOD_MS 300
+#define PM_BATTERY_SAMPLING_PERIOD_MS 100
+#define PM_SHUTDOWN_TIMEOUT_MS 15000
+#define PM_BATTERY_UNDERVOLT_THRESHOLD_V 3.0f
+#define PM_BATTERY_UNDERVOLT_HYSTERESIS_V 0.5f
+#define PM_BATTERY_LOW_THRESHOLD_V 3.15f
+#define PM_BATTERY_LOW_RECOVERY_V 3.2f
+#define PM_BATTERY_SAMPLING_BUF_SIZE 10
+
+#define PM_WPC_CHARGE_CURR_STEP_MA 50
+#define PM_WPC_CHARGE_CURR_STEP_TIMEOUT_MS 1000
+#define PM_FUEL_GAUGE_R 3000.0f
+#define PM_FUEL_GAUGE_Q 0.001f
+#define PM_FUEL_GAUGE_R_AGGRESSIVE 3000.0f
+#define PM_FUEL_GAUGE_Q_AGGRESSIVE 0.001f
+#define PM_FUEL_GAUGE_P_INIT 0.1f
+
+// Event flag manipulation macros
+#define PM_SET_EVENT(flags, event) ((flags) |= (event))
+#define PM_CLEAR_EVENT(flags, event) ((flags) &= ~(event))
+#define PM_CLEAR_ALL_EVENTS(flags) ((flags) = 0)
+
+/* Power manager states */
+#define PM_STATE_LIST(STATE) \
+  STATE(HIBERNATE)           \
+  STATE(CHARGING)            \
+  STATE(STARTUP_REJECTED)    \
+  STATE(SUSPEND)             \
+  STATE(ULTRA_POWER_SAVE)    \
+  STATE(SHUTTING_DOWN)       \
+  STATE(POWER_SAVE)          \
+  STATE(ACTIVE)
+
+typedef enum {
+#define STATE(name) PM_STATE_##name,
+  PM_STATE_LIST(STATE)
```

Comment:
as we are not using PM_STATE_LIST in different places, i'd go for simplification here and define the states directly